### PR TITLE
Add conversion to machine.cluster.k8s.io

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/api/conversions/conversions.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+	machinev1alpha1downstream "github.com/kubermatic/machine-controller/pkg/machines/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	clustercommon "sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+func ConvertV1alpha1DownStreamMachineToV1alpha1ClusterMachine(in machinev1alpha1downstream.Machine) (*clusterv1alpha1.Machine, error) {
```
Comment (p0lyn0mial) [inline pkg/api/conversions/conversions.go]: we could stick to how k8s defines conversion function, for example this is the method that will convert `StatefulSetSpec`

`Convert_v1beta1_StatefulSetSpec_To_apps_StatefulSetSpec(in *appsv1beta1.StatefulSetSpec, out *apps.StatefulSetSpec, s conversion.Scope) error`

so we could change our function to have the following signature ?
`Convert_v1alpha1_DownStreamMachine_To_v1alpha1_ClusterMachine(in *machinev1alpha1downstream.Machine, out *clusterv1alpha1.Machine) error`

If we don't like it then perhaps we could change our signature to accept and return a pointer to a struct ?
`func ConvertV1alpha1DownStreamMachineToV1alpha1ClusterMachine(in *machinev1alpha1downstream.Machine) (*clusterv1alpha1.Machine, error)`

File: pkg/api/conversions/conversions.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	// To work around this, we put it into the providerConfig
+	inMachineVersionJSON, err := json.Marshal(in.Spec.Versions)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal downstreammachine version: %v", err)
+	}
+	if err = json.Unmarshal(inMachineVersionJSON, &out.Spec.Versions); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal downstreammachine version: %v", err)
+	}
+
+	providerConfigMap, err := addContainerRuntimeInfoToProviderConfig(*out.Spec.ProviderConfig.Value,
```
Comment (p0lyn0mial) [inline pkg/api/conversions/conversions.go]: nit: we could return `[]byte` from `addContainerRuntimeInfoToProviderConfig` method

File: pkg/api/conversions/conversions.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+func addContainerRuntimeInfoToProviderConfig(providerConfigValue runtime.RawExtension, containerRuntimeInfo machinev1alpha1downstream.ContainerRuntimeInfo) (map[string]interface{}, error) {
+	providerConfigMap := map[string]interface{}{}
+	if err := json.Unmarshal(providerConfigValue.Raw, &providerConfigMap); err != nil {
+		return nil, fmt.Errorf("failed to unmarshall provider config into map: %v", err)
+	}
+	// The JSON unmarshall makes the map a null pointer if providerConfigValue.Raw is empty
+	if providerConfigMap == nil {
+		providerConfigMap = map[string]interface{}{}
+	}
+	providerConfigMap["containerRuntimeInfo"] = containerRuntimeInfo
```
Comment (p0lyn0mial) [inline pkg/api/conversions/conversions.go]: nice :) no longer part of `OperatingSytem` section.

**Comment by alvaroaleman** [comment]
Closing, this got implemented via #323

**Pull request closed without merging**